pkg/collector/corechecks/cluster/ksm/kubernetes_state_aggregators.go

```diff
@@ -135,6 +172,24 @@ func (a *countObjectsAggregator) accumulate(metric ksmstore.DDMetric) {
 	a.accumulator[labelValues]++
 }
 
+func (a *resourceAggregator) accumulate(metric ksmstore.DDMetric) {
+	var labelValues [maxNumberOfAllowedLabels]string
+
+	for i, allowedLabel := range a.allowedLabels {
+		if allowedLabel == "" {
+			break
+		}
+
+		labelValues[i] = metric.Labels[allowedLabel]
+	}
```

Comment:
Hi @sblumenthal, awesome work. 

I just have a small question. Is there a case where `allowedLabel` can be "" ? It looks like this field is set as `newResourceValuesAggregator` and it can be dynamically sized.

Also I think using `a.allowedLabels[i]` should save a copy operation.

Reply:
That is a good point, I think you are right. I was trying to mimic the way we are handling aggregation for the `sumValuesAggregator` and `countObjectsAggregator`, where the same loop is present, but thinking about it, I can not think of a situation where an allowed label WOULD be `""`, other than user error.

> Also I think using `a.allowedLabels[i]` should save a copy operation.

I don't think I quite understand what you are saying here, `allowedLabel` is being used as a key to get the value in `metric.Labels`. If you are suggesting that we do something like:
```
	for i := range a.allowedLabels {
		if a.allowedLabels[i] == "" {
			break
		}

		labelValues[i] = metric.Labels[a.allowedLabels[i]]
	}
```

Then I think that we could do something like that, but the reason against it is for readability and maintainability, but at a cost of a copy operation as you say.

Of course, the named second parameter in the range loop is consistent with the file, and with the codebase, so my question is whether this is a problem, and if it is, is it one that we want to start addressing?

Reply:
I am not sure how frequently this code is ran but I think limiting annotations could be valuable here. KSM core is already using a lot of memory and performing many allocations. Perhaps we could try to improve it.

Reply:
It is run up to 4 times per container and 4 times per node for each check, so it could be a lot on large clusters. Enough that I feel confident making the change is worth it.

I am going to leave the empty string check in for now though, just in case there is something happening during runtime in the aggregator codepath that I am overlooking